zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbResourceState.java

```diff
@@ -301,10 +290,21 @@ private PersistedResource getPersistedResourceById(
 
   private Optional<PersistedResource> getResourceFromCache(
       final String tenantId, final String resourceId) {
-    return Optional.ofNullable(
-        resourcesByTenantIdAndIdCache.getIfPresent(
-            new DbResourceState.TenantIdAndResourceId(tenantId, resourceId)));
+    try {
+      return Optional.of(
+          resourcesByTenantIdAndIdCache.get(new TenantIdAndResourceId(tenantId, resourceId)));
+    } catch (final ExecutionException e) {
+      // We reach this when we couldn't load the DRG from the state.
+      return Optional.empty();
+    }
   }
 
   private record TenantIdAndResourceId(String tenantId, String resourceId) {}
+
+  /**
+   * This exception is thrown when a DbResourceState cache can't find a PersistedResource in the
+   * state for the given parameters. This must be a checked exception, because of the way the {@link
+   * LoadingCache} works.
+   */
+  private static final class ResourceNotFoundException extends Exception {}
```

Comment:
:question: This appears unused?

Reply:
That's because in the rush to fix this, I forgot to use this! 😦 

I'll add a commit to use it properly.